src/com/company/Main.java

```diff
@@ -0,0 +1,76 @@
+package com.company;
+
+import com.company.faculty.Faculty;
+import com.company.faculty.FacultyFactory;
+import com.company.group.Group;
+import com.company.student.Student;
+import com.company.subject.English;
+import com.company.subject.Mathematics;
+import com.company.subject.TheoreticalComputerScience;
+import com.company.university.University;
+
+import java.util.Objects;
+
+public class Main {
+
+    public static void main(String[] args) {
+        try {
+            University university = University.getInstance();
+            System.out.println();
+            System.out.println("The university has two faculties: ");
+            Faculty mathematics = null;
+            try {
+                mathematics = FacultyFactory.getFaculty("Mathematics");
+                Group group1 = new Group("Group1");
+
+                mathematics.getGroups().add(group1);
+
+                try {
+                    Student student1 = new Student.SubjectBuilder()
+                            .addEnglish(8)
+                            .addHistory(4)
+                            .addMathematics(6)
+                            .addPhilosophy(7)
+                            .addTheoreticalComputerScience(5)
+                            .build();
+                    student1.setFirstName("Ashot ");
+                    student1.setLastName("Amaryan");
+                    group1.getStudents().add(student1);
```

Comment:
Lusine jan it is better to not throw exception on building student object, as in case of you want to have few students in group if one student will not be created(as it will not have  marks) others will not be created as well, as you will have cached exception in same block, or if you will have for each student each try/catch block  the code will become complex and unreadable, I will suggest throwing exceptions only on counting averages methods

Reply:
Diana jan, here  I want to precise something, if student.subject.size()==0 it doesn't mean that I have create a student, but I have not added subjects? And try/catch I have exactly for every student that I create, why other students will not to be created in the group? And if we have exception during the creation of the object we don't need to go farther to the method. What do you think? And thaaanks a lot for all your attention ^_^

Reply:
Lusine jan if you throw an exception in a builder, your particular student will not be created, so then you can't use it, in caller method fo example if you want to create two students and add them in the same group you will have for each student separate try/catch block, it will work, but in any case I think it is not right to not create student in case if it hasn't any subject, or someone tries to give not valid mark for some subject. But it may be right to have as requirement and specify on task giving time, we will make them more clear for next tasks, I hope :)